producer/producer_test.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright © 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package producer_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/Shopify/sarama/mocks"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/RedHatInsights/insights-results-aggregator/broker"
+	"github.com/RedHatInsights/insights-results-aggregator/producer"
+)
+
+// Test Producer creation with a non accesible Kafka broker
+func TestNewProducerBadBroker(t *testing.T) {
+	brokerCfg := broker.Configuration{
+		Address:      "localhost:1234",
+		PublishTopic: "topic",
+		Group:        "Group",
+	}
+
+	_, err := producer.New(brokerCfg)
+	assert.EqualError(t, err,
+		"kafka: client has run out of available brokers to talk to (Is your cluster reachable?)")
+}
+
+// Test ProduceMessage using a Sarama Mock producer. Asume sending success
+func TestProducerProduceMessage(t *testing.T) {
+	brokerCfg := broker.Configuration{
+		Address:      "localhost:1234",
+		PublishTopic: "topic",
+		Group:        "Group",
+	}
+
+	mockProducer := mocks.NewSyncProducer(t, nil)
+	mockProducer.ExpectSendMessageAndSucceed()
+	producer := producer.KafkaProducer{
+		Configuration: brokerCfg,
+		Producer:      mockProducer,
+	}
+
+	_, _, err := producer.ProduceMessage("Hello world")
+	assert.Equal(t, err, nil)
```

Comment:
In most of cases like this we use helpers.FailOnError(t, err) so it's better to follow one standard :)

Reply:
Ok! Thank you!